grift/loaders.py

```diff
@@ -67,6 +69,96 @@ def reload(self):
         self._source = os.environ
 
 
+class VaultException(Exception):
+    """Custom exception raised if Vault response contains error message"""
+
+
+class VaultTokenLoader(DictLoader):
+    def __init__(self, url, path, token):
+        """Load secrets from a Vault path, using token authentication
+
+        See https://www.vaultproject.io/docs
+
+        Args:
+            url: Vault url
+            path: Vault path to fetch secrets from
+            vault_token: token (must have access to vault path)
+        """
+        self._vault_url = url
+        self._path = path
+        self._token = token
+        source_dict = self._fetch_secrets()
```

Comment:
This pattern is problematic:
- most people would be surprised to learn that constructing a new object makes network requests (i.e. can raise IOErrors);
- network access makes it difficult to mock;
- it's generally bad practice to call `self` methods before the object has been fully initialized, since its state is by definition incomplete, and all such called methods (and methods called by them, and so on) need to mind how much of the object has been constructed up to that point.

I would strongly advise refactoring this into a function that performs all of these checks and network requests, and when done, constructs a loader object and returns it.